internal/controller/dynamicReconciler.go

```diff
@@ -0,0 +1,131 @@
+package controller
+
+import (
+	"context"
+	kumquatv1beta1 "kumquat/api/v1beta1"
+	"kumquat/repository"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type DynamicReconciler struct {
+	client.Client
```

Comment:
I can't figure out where we benefit from embedding this type, do we actually need to embed it or can we just do `Client client.Client` instead?